File: LAB QUIZZES/CSE221_Quiz04/_A4D/_A4D_Testing.py
import os, sys, ast, time, random, threading, subprocess
from collections import defaultdict
# Compile and run: pypy _A4D_Testing.py
def solve(N, edges):
    res = False
    degrees = {}
    for i in edges:
        u, v = i
        if u not in degrees.keys():
            degrees[u] = 0
        if v not in degrees.keys():
            degrees[v] = 0
        degrees[u] += 1
        degrees[v] += 1
    odd = 0
    for i in degrees.values():
        if i % 2 != 0:
            odd += 1
    if odd == 0 or odd == 2:
        return True
    print("odd:", odd)
    print(degrees)
    
    return res


# stdin
# 3 2
# 0 1
# 1 2

# N, M = map(int, input().split())
# edges = []
# for i in range(M):
#     u, v = map(int, input().split())
#     edges.append((u, v))

# # hardcoded input
#N = 5
#edges = [(0, 1), (0, 2), (0, 3), (2, 3), (2, 4), (3, 4)]
#N = 8
#edges = [(0, 2), (0, 4), (1, 0), (1, 4), (1, 6), (1, 5), (2, 4), (2, 5), (3, 2), (4, 3), (4, 7), (5, 3), (6, 4), (6, 2), (6, 5), (7, 0)]
N = 5
edges = [(0, 1), (0, 2), (0, 3), (0, 4), (1, 2), (1, 3), (2, 3), (2, 4), (3, 4)]
#start = time.time()
res = solve(N, edges)
#finish = time.time()
#elapsed = finish - start

# # file output
# with open("OutputOfYourCode.txt", "w") as fout:
#     fout.write(str(res))

# stdout
print(res)

#print("Execution time for the targeted one test is " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)

File: LAB QUIZZES/CSE221_Quiz04/_A4D/_A4D_Testing.java
import java.util.*;
import java.io.*;

// Compile and run: javac _A4D_Testing.java && java _A4D_Testing
public class _A4D_Testing {

    public static boolean solve(Integer N, ArrayList<_A4D_Solution.Pair<Integer, Integer>> edges) {
        boolean res = true;
        // Place your code here
        
        return res;
    }

    public static void main(String args[]) throws Exception {
        // stdin
        // 3 2
        // 0 1
        // 1 2
        // int N, M;
        // cin >> N >> M;
        // BufferedReader fin = new BufferedReader(new InputStreamReader(System.in));
        // String[] tempA = fin.readLine().split(" ");
        // int N = Integer.parseInt(tempA[0]);
        // int M = Integer.parseInt(tempA[1]);
        
        // ArrayList<_A4D_Solution.Pair<Integer, Integer>> edges = new ArrayList<>();
            
        // for(int i = 0; i < M; i++){
        //     tempA = fin.readLine().split(" ");
        //     int u = Integer.parseInt(tempA[0]);
        //     int v = Integer.parseInt(tempA[1]);
        //     edges.add(_A4D_Solution.Pair<Integer, Integer>(u, v));
        // }
        // fin.close();


        // // hardcoded input
        int N = 5;
        ArrayList<_A4D_Solution.Pair<Integer, Integer>> edges = new ArrayList<>();
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(0, 1));
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(0, 2));
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(0, 3));
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(2, 3));
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(2, 4));
        edges.add(new _A4D_Solution.Pair<Integer, Integer>(3, 4));
        

        long start = System.nanoTime();
        boolean res = solve(N, edges);
        long finish = System.nanoTime();
        long elapsed = finish - start;

        // // file output
        // BufferedWriter fout = new BufferedWriter(new
        // FileWriter("OutputOfYourCode.txt"));
        // fout.write(res.toString());
        // fout.close();

        // stdout
        System.out.println(res);

        System.out.printf("Execution time for the targeted one test is %.9fs\n", elapsed * 1e-9);
        System.exit(0);
    }
}
